internal/services/dynatrace/client/client.go

```diff
@@ -0,0 +1,24 @@
+package client
+
+import (
+	"fmt"
+
+	"github.com/hashicorp/go-azure-sdk/resource-manager/dynatrace/2023-04-27/monitors"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/common"
+)
+
+type Client struct {
+	MonitorClient *monitors.MonitorsClient
```

Comment:
Currently, for the client which is a thin wrapper of the underlying client, instead of having multiple underlying clients, we prefer defining it as an embedded field:

```suggestion
	*monitors.MonitorsClient
```